pkg/security/podsecuritypolicy/group/mustrunas.go

```diff
@@ -70,7 +70,7 @@ func (s *mustRunAs) Validate(_ *api.Pod, groups []int64) field.ErrorList {
 
 	for _, group := range groups {
 		if !s.isGroupValid(group) {
-			detail := fmt.Sprintf("%d is not an allowed group", group)
+			detail := fmt.Sprintf("group %d must be in the ranges: %v", group, s.ranges)
```

Comment:
@php-coder do we have a bug here?  Should it return early if it finds valid?  The scenario I'm thinking is that if you have multiple ranges defined and it matches on the second range won't it still return an error.  Please add a test for this and check the other ranged strategies.

We also don't seem to be testing that we get the expected number of errors on multiple non-matches.

Reply:
@pweil- Good catch! Look like a bug, indeed. Should it be fixed separately? In this case, it will be easier to backport it if needed and it won't block this PR. WDYT?

Reply:
Yes, it isn't directly related to the initial intent of this PR so I'm ok with an issue and follow up now that it's identified.

Reply:
@lnsp Lennart, we need your opinion here before proceed. Please, let us know what option is preferable to you:
1) you will fix the both issues as part of this PR
2) you will fix the both issues but as 2 separate PRs (in this case the 2nd will depend on 1st)
3) you work only on the current PR and I will fix this issue as part on another PR

Reply:
I would also leave it for another PR since it seems to be a bit out of scope.

Reply:
@lnsp Are you interested in working on it? (it should be simple and I could help on all the way ;)

Reply:
Sure, why not!

Reply:
@pweil- @lnsp I tried to reproduce a bug and couldn't. I looked again to the code and now I think that there is no bug. :)

> Should it return early if it finds valid?

This particular loop iterates over pod's supplementalGroups and we need to sure that all user provided groups are fall in in the allowed ranges.

Reply:
+1, after a second look I agree.  Thanks for following up.